src/AppInstallerCLICore/Commands/SettingsCommand.cpp

```diff
@@ -183,6 +184,21 @@ namespace AppInstaller::CLI
 
     void SettingsResetCommand::ValidateArgumentsInternal(Execution::Args& execArgs) const
     {
+        if (execArgs.Contains(Execution::Args::Type::All))
+        {
+            if (execArgs.Contains(Execution::Args::Type::SettingName))
+            {
+                throw CommandException(Resource::String::InvalidArgumentSpecifierError(ArgumentCommon::ForType(Execution::Args::Type::SettingName).Name));
+            }
+
+            return;

```

Comment:
if (ContainsAll && ContansSettingsName)
{
  throw ArgumentConflict()?
}

Reply:
Except that it also needs to return if not throwing.